ExampleMod/Common/GlobalItems/WeaponWithGrowingDamage.cs

```diff
@@ -144,15 +144,27 @@ public class WeaponWithGrowingDamage : GlobalItem
 	}
 	public class SnowBallShop : GlobalNPC
 	{
-		public override void SetupShop(int type, Chest shop, ref int nextSlot) {
-			if (type != ModContent.NPCType<ExamplePerson>()) {
+		public override void ModifyShop(NPCShop shop) {
+			if (shop.FullName != NPCShopDatabase.GetNPCShopName(ModContent.NPCType<ExamplePerson>(), "ExampliumShop")) {
 				return;
 			}
 
-			Item item = shop.item[nextSlot++];
-			item.SetDefaults(ItemID.Snowball);
-			if (item.TryGetGlobalItem(out WeaponWithGrowingDamage weapon)) {
-				weapon.GainExperience(item, 2); // can buy snowballs with 2xp!
+			var snowball = new Item(ItemID.Snowball);
+			if (snowball.TryGetGlobalItem(out WeaponWithGrowingDamage weapon)) {
+				weapon.GainExperience(snowball, 2);
+			}
+			shop.Add(snowball);
+		}
+
+		public override void ModifyActiveShop(NPC npc, string shopName, Item[] items) {
+			if (shopName != NPCShopDatabase.GetNPCShopName(ModContent.NPCType<ExamplePerson>(), "ExampliumShop")) {
+				return;
+			}
+
+			foreach (Item item in items) {
+				if (item.type == ItemID.Snowball && item.TryGetGlobalItem(out WeaponWithGrowingDamage weapon)) {
+					weapon.GainExperience(item, 2); // can buy snowballs with 2xp!
+				}
```

Comment:
This code seems unnecessary? A bad port? You already add snowballs with 2xp to the shop in `ModifyShop`. This would increase their `xp` to 4